src/index.ts

```diff
@@ -629,7 +632,22 @@ class ThreadPool {
     this.workers.delete(workerInfo);
   }
 
+  // On Windows, os.loadavg() always returns [0,0,0], so
+  // this check does not make sense on that platform, but
+  // we go ahead and check any way.
+  _acceptableCpuLoad (this: ThreadPool) : boolean {
+    if (this.workers.size < this.options.minThreads ||
+        this.options.cpuLoadAvgThreshold === Infinity) {
+      return true;
+    }
+    return loadavg()[0] / cpuCount <= this.options.cpuLoadAvgThreshold;
+  }
+
   _onWorkerAvailable (workerInfo : WorkerInfo) : void {
+    // We do not check _acceptableCpuLoad here with the idea
+    // that if a worker is coming available, then it should
+    // always be permitted to take on a new task if one is
+    // available.
```

Comment:
I’m not sure whether it makes sense to distinguish this case from the one below?